astropy/io/fits/src/compressionmodule.c

```diff
@@ -1061,6 +1061,24 @@ PyObject* compression_decompress_hdu(PyObject* self, PyObject* args)
     if (PyErr_Occurred()) {
         return NULL;
     }
+    // non-integer scaling or subtractive dithering means floating-point
+    if ((long)fileptr->Fptr->cn_bscale != fileptr->Fptr->cn_bscale ||
```

Comment:
Is that how one tests for floats in `C`? It's probably correct but it seems weird.

Reply:
Should be.  The same implementation is used in cfitsio.

Reply:
okay, I was thinking about possible overflow when casting `float` to `signed long integer`. I never know if that's just implementation defined or undefined.

Reply:
I'm not a C programmer so if you say that's okay I'll trust you :)